src/monitor/monitor.c

```diff
@@ -1642,6 +1642,12 @@ errno_t load_configuration(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
+    /* return EOK so genconf=KCM may start without any
+     * sssd configuration */
+    if (only_section != NULL) {
```

Comment:
In theory, this might be different request, not necessarily KCM, right? Are we sure this approach is OK regardless what section was requested?

Reply:
Correct, as far as I can tell in SSSD sources `--genconf-section` is only used for KCM:
~~~
src/sysv/systemd/sssd-kcm.service.in:12:ExecStartPre=-@sbindir@/sssd --genconf-section=kcm
~~~

I see it is also used in ``src/tests/multihost/basic/test_config.py`` I will run this to make sure this PR doesn't break this test.

Reply:
I updated the comment and added a check to ensure genconf-section == KCM also. Will wait for CI to check ``src/tests/multihost/basic/test_config.py` is successful in multihost-pytest.log

Reply:
Thank you.

Reply:
Actually... If `(only_section != NULL)` but it's not `"kcm"` then `talloc_free(ctx);` in `done` section is unnecessary.

Reply:
Okay understood, thank you. I removed the strcmp condition.